packages/cozy-sharing/src/components/ShareDialogCozyToCozy.jsx

```diff
@@ -32,76 +159,148 @@ const ShareDialogCozyToCozy = ({
   showShareByEmail,
   showShareByLink,
   showShareOnlyByLink,
-  showWhoHasAccess
+  showWhoHasAccess,
+  hasTwoStepsConfirmation = false,
+  twoStepsConfirmationMethods = {}
 }) => {
   const { t } = useI18n()
+
+  const [status, setStatus] = React.useState('loading')
+  const [recipientsToBeConfirmed, setRecipientsToBeConfirmed] = useState([])
+  const [recipientConfirmationData, setRecipientConfirmationData] = useState(
+    undefined
+  )
+
+  const {
+    confirmRecipient,
+    rejectRecipient,
+    recipientConfirmationDialogContent: ConfirmationDialogContent
+  } = twoStepsConfirmationMethods
+
+  const mountedRef = React.useRef(false)
+  useEffect(() => {
+    mountedRef.current = true
+
+    return () => {
+      mountedRef.current = false
+    }
+  })
```

Comment:
You do that in order to know if the modal is still opened, right? Wondering if we should not provide a useSafeState hook like https://github.com/tannerlinsley/react-query/blob/79ad5a9ccbce1486c10e7e364d8fc28bef3fa19f/src/devtools/utils.ts#L80 for instance

Reply:
You guessed it right.

Maybe this will be an opportunity to initiate the Hook library we mentioned in `Tech Transverse` board.

Reply:
I implemented a first hook in cozy-sharing so I can use it without waiting for the new package to be created.

Related commit : https://github.com/cozy/cozy-libs/pull/1340/commits/86ddb21d7d4ffebb5f322749d225017f6b9a2112

Later I expect to create this package and edit current code to use the new hook from this project.